pdr_backend/pdr_dashboard/pages/feeds.py

```diff
@@ -1,22 +1,128 @@
 
 
 class FeedsPage:
+    def __init__(self, app):
+        self.app = app
+
     def layout(self):
         return html.Div(
             [
                 dcc.Store(id="user-payout-stats"),
-                html.H1(
-                    "Feeds page content",
-                    id="page_title_feeds",
-                ),
                 dcc.Loading(
                     id="loading",
                     type="default",
-                    children=[],
+                    children=[
+                        self.get_main_container(),
+                    ],
                     custom_spinner=html.H2(dbc.Spinner(), style={"height": "100%"}),
                 ),
             ],
             style={"height": "100%"},
         )
+
+    def get_main_container(self):
+        feed_stats = get_feed_payouts_stats_from_db(self.app.lake_dir)
+        feed_subscriptions = get_feed_subscription_stats_from_db(
+            self.app.lake_dir, self.app.network_name
+        )
+
+        feed_cols, feed_data = self.get_feeds_data_for_feeds_table(
+            feed_stats, feed_subscriptions
+        )
+
+        return html.Div(
+            [self.get_feeds_table_area(feed_cols, feed_data)],
+            className="main-container",
+        )
+
+    def get_feeds_table_area(
+        self,
+        columns,
+        feeds_data,
+    ):
+        return html.Div(
+            [
+                dash_table.DataTable(
+                    id="feeds_table",
+                    columns=columns,
+                    data=feeds_data,
+                    sort_action="native",  # Enables sorting feature
+                )
+            ],
+            style={"width": "100%"},
+        )
+
+    def get_feeds_stat_with_contract(
+        self, contract: str, feed_stats: List[Dict[str, Any]]
+    ) -> Union[Tuple[float, float, float], None]:
+        result = find_with_key_value(feed_stats, "contract", contract)
+
+        if result:
+            return (
+                round(result["avg_accuracy"], 2),
+                round(result["volume"], 2),
+                round(result["avg_stake"], 2),
+            )
+
+        return None
+
+    def get_feeds_data_for_feeds_table(
+        self,
+        feed_stats: List[Dict[str, Any]],
+        feed_subcription_stats: List[Dict[str, Any]],
+    ) -> Tuple[List[Dict[str, str]], List[Dict[str, Any]]]:
+
+        temp_data = self.app.feeds_data
+
+        new_feed_data = []
+        ## split the pair column into two columns
+        for feed in temp_data:
+            split_pair = feed["pair"].split("/")
+            feed_item = {}
+            feed_item["addr"] = feed["contract"][:5] + "..." + feed["contract"][-5:]
+            feed_item["base_token"] = split_pair[0]
+            feed_item["quote_token"] = split_pair[1]
+            feed_item["exchange"] = feed["source"].capitalize()
+            feed_item["time"] = feed["timeframe"]
+
+            result = self.get_feeds_stat_with_contract(feed["contract"], feed_stats)
```

Comment:
I would suggest returning a dictionary here instead of a tuple, and simply updating the initial dictionary with the feed stat dictionary. Why rely on tuple orders if you can use the same data structure as the caller?

Reply:
I fixed it, and also created another method for the subscription stats (`get_feeds_subscription_stat_with_contract`) and used in this method, for more clear code.